tasksCode/StringList.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+	Set of the interface functions for using the dynamic list of c strings. Done
+	on the SoftServe' demand. According to the convention the functions starting
+	with underline, e.g. <_FuncName>, are hidden and as they don't relate to the
+	interface should not be used. Exception type - const char *.
+	Author:						Valentyn Faychuk
+	E-mail:						faitchouk.valentyn@gmail.com
+	Created on:					06.07.2019
+*/
+#include <stdlib.h>
+#include <string.h>
+
+// Type for the string itself
+typedef char * String;
+// Types for the list size as well as string sizes
+typedef int StringListSizeType;
+typedef int StringSizeType;
+
+const char * g_badAllocMessage = "Bad memory allocation";
+const char * g_NULLPtrParameterMessage = "nullptr as a parameter";
+
+// ******************************************************************* CONTROLLERS
+
+// Initialize list
+void StringListInit(char *** list);
+
+// Destroy list and set pointer to the nullptr
+void StringListDestroy(char *** list);
+
+// ******************************************************************* /CONTROLLERS
+
+// ************************************************************************ GETTERS
+
+// Return the number of items in the list
+int StringListSize(char ** list);
+
+// Give the allocation size of the list
+StringListSizeType _StringListAllocatedSize(char ** list);
+
+// Return the index position of the first occurrence of the str in the list
+int StringListIndexOf(char ** list, const char * str);
+
+// Return the index position of the first occurrence of the str in the list
+// Start its search from given index if the start parameter is present
+StringListSizeType _StringListIndexOf(char ** list, const char *, StringListSizeType start);
+
+// ************************************************************************ /GETTERS
+
+// ************************************************************************ MUTATORS
+
+// Insert the value at the end
+void StringListAdd(char *** list, const char * str);
```

Comment:
Not suitable interface.

Reply:
I suppose there is mismatch in the requirements probably. The string addition to the list mean that we have to change list allocation dynamically. I.e. this action doesn't mutate elements, but collection.
That's why I have chosen to pass my list by pointer here.